processor/src/chiplets/aux_trace/virtual_table.rs

```diff
@@ -175,3 +243,77 @@ where
         E::ONE
     }
 }
+
+// LOG PRECOMPILE MESSAGES
+// ================================================================================================
+
+/// Message for log_precompile transcript-state tracking on the virtual table bus.
+struct LogPrecompileMessage {
+    state: PrecompileTranscriptState,
+}
+
+impl<E> BusMessage<E> for LogPrecompileMessage
+where
+    E: FieldElement<BaseField = Felt>,
+{
+    fn value(&self, alphas: &[E]) -> E {
+        let state_elements: [Felt; 4] = self.state.into();
+        alphas[0]
+            + alphas[1].mul_base(Felt::from(LOG_PRECOMPILE_LABEL))
+            + build_value(&alphas[2..6], state_elements)
+    }
+
+    fn source(&self) -> &str {
+        "log_precompile"
+    }
+}
+
+impl core::fmt::Display for LogPrecompileMessage {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        write!(f, "{{ state: {:?} }}", self.state)
+    }
+}
+
+/// Removes the previous transcript state (`CAP_PREV`) from the virtual table bus.
+///
+/// Helper register layout for `log_precompile` is codified as:
+/// - `h0` = hasher address, `h1..h4` = `CAP_PREV[0..3]`.
+fn build_log_precompile_capacity_remove<E: FieldElement<BaseField = Felt>>(
+    main_trace: &MainTrace,
+    row: RowIndex,
+    alphas: &[E],
+    _debugger: &mut BusDebugger<E>,
+) -> E {
+    // The previous transcript state is the capacity word provided non-deterministically in the
+    // helper registers, offset by 1 to account for the hasher address
+    let state: PrecompileTranscriptState =
+        [1, 2, 3, 4].map(|idx| main_trace.helper_register(idx, row)).into();
+
+    let message = LogPrecompileMessage { state };
+    let value = message.value(alphas);
+
+    #[cfg(any(test, feature = "bus-debugger"))]
+    _debugger.add_request(alloc::boxed::Box::new(message), alphas);
+
+    value
+}
```

Comment:
The hardcoded indices `[11, 10, 9, 8]` should be replaced with named constants or derived from a base offset to improve maintainability.

For example:
```rust
const CAPACITY_BASE_OFFSET: usize = 8;
const WORD_LEN: usize = 4;

let capacity_indices: [usize; WORD_LEN] = [
    CAPACITY_BASE_OFFSET + WORD_LEN - 1,
    CAPACITY_BASE_OFFSET + WORD_LEN - 2,
    CAPACITY_BASE_OFFSET + WORD_LEN - 3,
    CAPACITY_BASE_OFFSET + WORD_LEN - 4,
];

let capacity: Word = capacity_indices
    .map(|idx| main_trace.stack_element(idx, row + 1)).into();
```

This would make it clearer that we're accessing the CAP_NEXT word at stack indices 8-11 in reverse order.